promql/lex.go

```diff
@@ -86,6 +86,14 @@ func (i itemType) isSetOperator() bool {
 	return false
 }
 
+// isLabel returns whether the item can be used as label.
+func (i itemType) isLabel() bool {
+	if i == itemIdentifier || i.isOperator() || i.isAggregator() || i.isKeyword() {
```

Comment:
Sure thing! And you're right, this is too broad.

It's probably easier to just validate that the value matches our label format.